File: replace.py
import os

with open('app_privee.html', 'r', encoding='utf-8') as f:
    html = f.read()

with open('../brain/372eee46-324a-4111-af73-8f37147edce4/scratch/new_pane3.html', 'r', encoding='utf-8') as f:
    new_pane3 = f.read()

with open('../brain/372eee46-324a-4111-af73-8f37147edce4/scratch/new_js.js', 'r', encoding='utf-8') as f:
    new_js = f.read()

start_index = html.find('<!-- FORMULAIRE 3 : Audit Devis')
if start_index == -1:
    print("Could not find start index of Formulaire 3")
    exit(1)

end_index = html.find('<!-- FORMULAIRE 4', start_index)
if end_index == -1:
    print("Could not find end index of Formulaire 3")
    exit(1)

new_html = html[:start_index] + new_pane3 + "\n\n        " + html[end_index:]

js_start_index = new_html.find('window.addDevisLine = function()')
if js_start_index == -1:
    print("Could not find js start index")
    exit(1)

js_end_index = new_html.find('// 8. LOGIQUE GLOBALE DE PAIEMENT', js_start_index)
if js_end_index == -1:
    js_end_index = new_html.find('</script>', js_start_index)

actual_js_start = new_html.rfind('let devisLineCount', 0, js_start_index)
if actual_js_start != -1:
    js_start_index = actual_js_start

new_html = new_html[:js_start_index] + new_js + "\n" + new_html[js_end_index:]

with open('app_privee.html', 'w', encoding='utf-8') as f:
    f.write(new_html)

print("Replacement done.")

File: replace.js
const fs = require('fs');
const html = fs.readFileSync('app_privee.html', 'utf8');
const newPane3 = fs.readFileSync('../brain/372eee46-324a-4111-af73-8f37147edce4/scratch/new_pane3.html', 'utf8');
const newJs = fs.readFileSync('../brain/372eee46-324a-4111-af73-8f37147edce4/scratch/new_js.js', 'utf8');

let startIndex = html.indexOf('<!-- FORMULAIRE 3 : Audit Devis BTP -->');
if (startIndex === -1) {
  startIndex = html.indexOf('<!-- FORMULAIRE 3 : Audit Devis');
}
let endIndex = html.indexOf('<!-- FORMULAIRE 4 : FINITIONS', startIndex);
if(endIndex === -1) {
  endIndex = html.indexOf('<!-- FORMULAIRE 4', startIndex);
}

let newHtml = html.substring(0, startIndex) + newPane3 + "\n\n        " + html.substring(endIndex);

let jsStartIndex = newHtml.indexOf('window.addDevisLine = function()');
let jsEndIndex = newHtml.indexOf('// 8. LOGIQUE GLOBALE DE PAIEMENT');
if (jsEndIndex === -1) jsEndIndex = newHtml.indexOf('</script>'); // fallback

// The old JS starts around "let devisLineCount = 0;"
let actualJsStart = newHtml.lastIndexOf('let devisLineCount', jsStartIndex);
if(actualJsStart !== -1) jsStartIndex = actualJsStart;

newHtml = newHtml.substring(0, jsStartIndex) + newJs + "\n" + newHtml.substring(jsEndIndex);

fs.writeFileSync('app_privee.html', newHtml, 'utf8');
console.log('Replacement done.');
